crates/engine/src/websocket.rs

```diff
@@ -39,24 +36,34 @@ pub struct SubscribeEvent {
 pub struct RequestPayload(pub(crate) Request);
 
 #[derive(Debug, Default, serde::Deserialize)]
-pub struct InitPayload {
-    #[serde(default, deserialize_with = "deserialize_as_hash_map")]
-    pub headers: http::HeaderMap,
-}
+pub struct InitPayload(pub(crate) serde_json::Map<String, serde_json::Value>);
+
+impl InitPayload {
+    pub(crate) fn to_headers(&self) -> http::HeaderMap {
```

Comment:
This is best effort. Maybe trying a bit too hard, and too optimistic (ignores errors). I'm very open to better ideas.

Reply:
I just wonder how useful it is, sounds like something that would be worth dedicated rules/hook when asked for rather than trying to guess headers

Reply:
I'm happy with completely removing that part of the logic, and passing empty headers to the request context constructor.

Reply:
If that sounds best.

Reply:
Not entirely sure to be honest, but it just feels arbitrary. Think we should probably just treat them separately and provide similar looking rules/hooks in the future to better handle it. Like `on_gateway_http_request` & `on_gateway_websocket_request` for example. So would remove it yeah